packages/hackmcx-api/src/controllers/users.controller.js

```diff
@@ -21,3 +21,46 @@ export async function getUsersByUserId(req, res){
             }
         })
 }
+
+export async function postUser(req, res){
+    const emptyUsername = !req.body.username || /^\s*$/.test(req.body.username);
+    const emptyFirstName = !req.body.firstname || /^\s*$/.test(req.body.firstname);
+    const emptyLastName = !req.body.lastname || /^\s*$/.test(req.body.lastname);
+    const usernameExists = (await dbClient.select('username').from('users').where('username', req.body.username)).length > 0 ? true : false;
+    
+    const noGivenImageUrl = !req.body.imageUrl;
+    const notPicture = !req.body.imageUrl || !/(https?:\/\/.*\.(?:png|jpg|gif|svg))/i.test(req.body.imageUrl);
+
+    if(emptyUsername){
```

Comment:
We want to do all the validation checks we can before making a database query for this short circuit style evaluation. This can save a trip to the DB when the request is invalid anyway.